cmd/deploy/deploy.go

```diff
@@ -286,18 +287,28 @@ func (dc *DeployCommand) RunDeploy(ctx context.Context, deployOptions *Options)
 		return oktetoErrors.ErrDeployCantDeploySvcsIfNotCompose
 	}
 
+	// We need to create a client that doesn't go through the proxy to create
+	// the configmap without the deployedByLabel
+	c, _, err := dc.K8sClientProvider.Provide(okteto.Context().Cfg)
+	if err != nil {
+		return err
+	}
+
 	cwd, err := os.Getwd()
 	if err != nil {
 		return fmt.Errorf("failed to get the current working directory: %w", err)
 	}
 
-	// We need to create a client that doesn't go through the proxy to create
-	// the configmap without the deployedByLabel
-	c, _, err := dc.K8sClientProvider.Provide(okteto.Context().Cfg)
+	topLevelGitDir, err := repository.FindTopLevelGitDir(cwd, afero.NewOsFs())
 	if err != nil {
-		return err
+		oktetoLog.Warning("Some git env vars might not be available: %s", err.Error())
+	}
+
+	if topLevelGitDir != "" {
+		dc.addEnvVars(topLevelGitDir)
```

Comment:
decided to keep the call to `addEnvVars` twice, I've considered having a string defining the path, but it would make the code less readable IMHO